ee/tables/execparsers/data_table/parser.go

```diff
@@ -0,0 +1,116 @@
+package data_table
+
+import (
+	"bufio"
+	"io"
+	"strings"
+)
+
+// data_table is a general parser for an input of data which conforms to columns and rows styled output.
+// Parser options
+// skipLines - The number of initial lines of data to skip. By default no lines are skipped. This can be useful if consistent undesired output/garbage is printed before the data to parse.
+// headers - The set of headers. If left blank, the parser assumes the headers are in the first line of data and splits that line to set them.
+// delimiter - The splitting string. If left blank, the parser assumes the delimiter is whitespace and uses `strings.Fields()` split method.
+type parser struct {
+	skipLines uint
+	headers   []string
+	delimiter string
+}
+
+type parserOpt func(*parser)
+
+func WithSkipLines(skipLines uint) parserOpt {
+	return func(p *parser) {
+		p.skipLines = skipLines
+	}
+}
+
+func WithHeaders(headers []string) parserOpt {
+	return func(p *parser) {
+		p.headers = headers
+	}
+}
+
+func WithDelimiter(delimiter string) parserOpt {
+	return func(p *parser) {
+		p.delimiter = delimiter
+	}
+}
+
+func NewParser(opts ...parserOpt) *parser {
+	p := &parser{}
+
+	for _, opt := range opts {
+		opt(p)
+	}
+
+	return p
+}
+
+func (p parser) Parse(reader io.Reader) (any, error) {
+	return p.parseLines(reader)
+}
+
+// parseLines scans a reader line by line and splits it into fields based on a delimiter.
+// The line fields are paired with a header, which is defined by an input array, or the first line of data.
+func (p parser) parseLines(reader io.Reader) ([]map[string]string, error) {
+	results := make([]map[string]string, 0)
+	scanner := bufio.NewScanner(reader)
+
+	// Skip first N lines due to provided headers or otherwise.
+	// This would likely only ever be 1 or 0, but we may want more.
+	for p.skipLines > 0 {
+		p.skipLines--
+		// Early exit if the scanner skipped past all data.
+		if !scanner.Scan() {
+			return results, nil // <-- Do we want to error here?
```

Comment:
Should this comment be removed/resolved?

Reply:
This would occur if the data is smaller than the `skipLines`? If we end up in the case, how would we know and debug it?

If we return nil here, we don't have much indication of the issue. If we error out, it should get logged by the caller, right? So my inclination is to return an error